rotkehlchen/chain/ethereum/modules/uniswap/v2/common.py

```diff
@@ -120,9 +125,15 @@ def decode_uniswap_v2_like_swap(
             ) and
             event.asset == A_ETH and transaction.from_address == event.location_label
         ):
-            # Those are assets returned due to a change in the swap price
-            event.event_type = HistoryEventType.TRANSFER
-            event.counterparty = counterparty
-            event.notes = f'Refund of {event.balance.amount} {crypto_asset.symbol} in {counterparty} due to price change'  # noqa: E501
+            # this is to make sure its the amm issuing the refund and not an agreggator making a swap  # noqa: E501
+            if event.counterparty in (SUSHISWAP_ROUTER, UNISWAP_V2_ROUTER):
+                # Those are assets returned due to a change in the swap price
+                event.event_type = HistoryEventType.TRANSFER
+                event.counterparty = counterparty
+                event.notes = f'Refund of {event.balance.amount} {crypto_asset.symbol} in {counterparty} due to price change'  # noqa: E501
+            else:
+                event.event_type = HistoryEventType.TRADE
+                event.event_subtype = HistoryEventSubType.RECEIVE
```

Comment:
Shouldn't counterparty be set here too?

Reply:
The counterparty isn't set there because... the said event isn't supported by i.e. **CowSwap** so it makes no sense to put a counterparty.